From d6917964154bcd31cd21b75775d11fd6e695c153 Mon Sep 17 00:00:00 2001
From: minicx <minicx@disroot.org>
Date: Sun, 13 Jul 2025 12:07:13 +0300
Subject: [PATCH 1/1] buildcommands: Retrieve version from local file if Git version is unavailable

Updated the version retrieval logic in buildcommands.py to use a local version file when the Git version cannot be obtained.

Signed-off-by: minicx <minicx@disroot.org>
---
 scripts/buildcommands.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/buildcommands.py b/scripts/buildcommands.py
index b3587384..e7184956 100644
--- a/scripts/buildcommands.py
+++ b/scripts/buildcommands.py
@@ -7,7 +7,7 @@
 import sys, os, subprocess, optparse, logging, shlex, socket, time, traceback
 import json, zlib
 sys.path.append('./klippy')
-import msgproto
+import msgproto, util
 
 FILEHEADER = """
 /* DO NOT EDIT! This is an autogenerated file. See scripts/buildcommands.py. */
@@ -486,8 +486,7 @@ def git_version():
 def build_version(extra, cleanbuild):
     version = git_version()
     if not version:
-        cleanbuild = False
-        version = "?"
+        version = util.get_version_from_file('./klippy')
     elif 'dirty' in version:
         cleanbuild = False
     if not cleanbuild:
-- 
2.44.2

